test/server/cards/01_SOR/events/PowerFailure.spec.ts

```diff
@@ -0,0 +1,65 @@
+describe('Power Failure', function() {
+    integration(function(contextRef) {
+        describe('Power Failure\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['power-failure'],
+                        groundArena: [{ card: 'pyke-sentinel', upgrades: ['entrenched', 'devotion'] }],
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                        spaceArena: [{ card: 'imperial-interceptor', upgrades: ['academy-training', 'shield'] }]
+                    }
+                });
+            });
+
+            it('defeats all upgrades on a unit', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.powerFailure);
+                expect(context.player1).toBeAbleToSelectExactly([context.pykeSentinel, context.imperialInterceptor]);
```

Comment:
I think I mentioned this in my comment about setting `optional: true`, but we should add a test here confirming that the pass button appears: `expect(context.player1).toHavePassAbilityPrompt();`